File: 002-mockowanie/PHP/src/Util/Email.php
<?php
declare(strict_types=1);

namespace App\Util;

use \Exception;

/**
 * Class Email
 * @package App\Util
 */
class Email implements EmailInterface
{
    /**
     * @param string $email
     * @return bool
     * @throws Exception
     */
    public function send(string $email): bool
    {
        if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
            throw new Exception('Email is not valid');
        }

        //Send email

        return true;
    }
}

File: 002-mockowanie/Java/src/main/java/lukaszstaniszewski/util/Email.java
package lukaszstaniszewski.util;

/**
 * The type Email.
 */
public class Email implements EmailInterface {

    @Override
    public boolean send(String email) throws Exception {
        if (!email.matches("^([a-z|A-Z|0-9]){1,24}@([a-z|A-Z|0-9]){1,64}.([a-z]){2,6}$")) {
            throw new Exception("Email is invalid");
        }

        // Send mail

        return  true;
    }
}
